dv-export/python/query-changes/command_handler.py

```diff
@@ -30,32 +47,48 @@ def execute_export_job(self, base_data_version: int, data_version: int) -> str:
+        filter_name = self._get_filter_name(query)
+        filter_values = self._read_filter_values(export_uuid, analytic_object, filter_name) if filter_name else None
 
         # Retrieve analytic object changes
-        properties, changes_rows = self.changes_fetcher.list_changes(query, filter_values)
+        analytic_object_metadata, properties = self.changes_fetcher.get_analytic_object_metadata(analytic_object)
+
+        # Getting start and end for query changes depending on command mode
+        start_time_epoch, end_time_epoch = self._get_time_period_borders(export_uuid,
+                                                                         analytic_object_metadata,
+                                                                         query_mode)
+
+        # Fetch analytic object changes
+        changes_rows = self.changes_fetcher.execute_query_list(query, start_time_epoch, end_time_epoch,
+                                                               self.record_modes[query_mode], filter_values)
 
         # Save changes to the database
-        self._save_to_db(analytic_object, query, properties, changes_rows)
-
-    def _read_filter_values(self, analytic_object_name: str, export_uuid: str, query):
-        filter_property_name = self._get_filter_property_name(query)
-        filter_values = None
-        if filter_property_name is not None:
-            # Get filter values from export files
-            property_values = self.dv_manager.read_property_values(export_uuid, analytic_object_name,
-                                                                   filter_property_name)
-            filter_values = set(property_values)
+        self._save_to_db(analytic_object, filter_name, filter_values, query, properties, changes_rows, query_mode)
+
+    def _get_time_period_borders(self, export_uuid: str, analytic_object_metadata: dict[str, typing.Any],
+                                 query_mode: QueryMode) -> (int, int):
+        if query_mode == QueryMode.LAST:
+            start_time_epoch, end_time_epoch = self.dv_manager.get_export_data_version_times(export_uuid)
+        elif query_mode == QueryMode.RESTATE:
+            start_time_epoch, end_time_epoch = int(analytic_object_metadata[DATA_START_DATE]), int(time.time() * 1000)
+        else:
+            raise ValueError(f"Unknown command mode: {query_mode}")
+        return start_time_epoch, end_time_epoch
+
+    def _read_filter_values(self, export_uuid: str, analytic_object_name: str, filter_name: str):
+        # Get filter values from export files
+        property_values = self.dv_manager.read_property_values(export_uuid, analytic_object_name, filter_name)
+        filter_values = set(property_values)
         return filter_values
 
     @staticmethod
-    def _get_filter_property_name(query: dict[str, typing.Any]):
+    def _get_filter_name(query: dict[str, typing.Any]) -> str | None:
```

Comment:
should we be more consistent with function return type hints?

Reply:
Updated type annotations.